File: Backend/src/main/java/com/example/bus_reservation_system/entity/Ticket.java
package com.example.bus_reservation_system.entity;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
@Table(name = "ticket")
public class Ticket {


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

//    @Column(nullable = false)
    private long reservation_id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String gender;

    @Column(nullable = false)
    private int age;

    @Column(nullable = false)
    private String stateOfResidence;

    @Column(nullable = false)
    private String contactDetails;

    @Column(nullable = false)
    private boolean donation;
}

File: Frontend/src/app/Models/ticket.ts
export class Ticket {
    id?: number;
    reservation_id?: number;
    name: string;
    age: number;
    gender: string;
    stateOfResidence: string;
    contactDetails: string;
    
    constructor(
      name: string='',
      age: number=0,
      gender: string='',
      state_of_residence: string='',
      contact_details: string='',
      id?: number,
      reservation_id?: number
    ) {
      this.name = name;
      this.age = age;
      this.gender = gender;
      this.stateOfResidence = state_of_residence;
      this.contactDetails = contact_details;
      this.id = id; // Optional
      this.reservation_id = reservation_id; // Optional
    }
  }
  